examples/cpp/helloworld/CMakeLists.txt

```diff
@@ -55,6 +55,23 @@ if(GRPC_AS_SUBMODULE)
   set(_PROTOBUF_PROTOC $<TARGET_FILE:protoc>)
   set(_GRPC_GRPCPP_UNSECURE grpc++_unsecure)
   set(_GRPC_CPP_PLUGIN_EXECUTABLE $<TARGET_FILE:grpc_cpp_plugin>)
+elseif(GRPC_FETCHCONTENT)
+  # Another way is to use CMake's FetchContent module to clone gRPC at
+  # configure time. This makes gRPC's source code available to your project,
+  # similar to a git submodule.
+  include(FetchContent)
+  FetchContent_Declare(
+    grpc
+    GIT_REPOSITORY https://github.com/grpc/grpc.git
+    GIT_TAG        v1.25.0)
+  FetchContent_MakeAvailable(grpc)
+
+  # Since FetchContent uses add_subdirectory under the hood, we can use

```

Comment:
IMHO This is not exactly straightforward and requires the user to know the internals of the gRPC's CMakeLists.txt which kind of beats the purpose of trying to make gRPC C++ easy to use with cmake.